src/en/coloredmanga/src/eu/kanade/tachiyomi/extension/en/coloredmanga/ColoredManga.kt

```diff
@@ -0,0 +1,417 @@
+                }
+            }
+        }
+
+        val final = sorted.map(::popularManga)
+        return MangasPage(final, false)
+    }
+
+    private fun getMangas(doc: Document): List<Mangas> {
+        val mangasRaw = doc.selectFirst("script:containsData(\\\"cover)")!!.data()
+        val mangasJson = mangasRaw
+            .replace(Regex("""\\([\\"])"""), "$1")
+            .replaceBefore("\"data\":[", "{")
+            .removeSuffix("]}]\\n\"])")
+
+        return mangasJson.parseAs<MangasList>().data
+    }
+
+    private fun getManga(url: String): Request {
+        val formData = MultipartBody.Builder().setType(MultipartBody.FORM)
+            .addFormDataPart("id", url.substringAfter("manga/"))
+            .build()
+
+        val request = Request.Builder()
+            .url("$baseUrl/api/selectedManga")
+            .put(formData)
+            .addHeader("Cache-Control", "no-store")
+            .build()
+
+        return request
+    }
+
+    private fun getImage(manga_name: String, volume_name: String? = "", chapter: Chapter, index: String): String {
+        val chapterNumber = index.padStart(4, '0')
+        val chapterName = listOf(chapter.number, chapter.title)
+            .joinToString(" - ")
+            .trim()
+            .replace(nameRegex, "")
+
+        val formData = MultipartBody.Builder().setType(MultipartBody.FORM)
+            .addFormDataPart("path", "/images/content/$manga_name/$volume_name$chapterName")
+            .addFormDataPart("number", chapterNumber)
+            .build()
+
+        val request = Request.Builder()
+            .url("$baseUrl/api/dynamicImages")
+            .put(formData)
+            .addHeader("Cache-Control", "no-store")
+            .build()
+
+        val response = client.newCall(request).execute().body.string()
+
+        val responseJson = JSONObject(response)
+        val image = responseJson.getString("image")
+
+        return image
+    }
+
+    private fun popularManga(manga: Mangas): SManga {
+        return SManga.create().apply {
+            title = manga.name
+            setUrlWithoutDomain("$baseUrl/manga/${manga.id}")
+            thumbnail_url = "$baseUrl${manga.cover}"
+        }
+    }
+
+    // Latest
+    override fun latestUpdatesRequest(page: Int): Request = popularMangaRequest(page)
+
+    override fun latestUpdatesParse(response: Response): MangasPage = searchMangas(response, sortBy = "lat" to "desc")
+
+    // Search
+
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList,
+    ): Observable<MangasPage> {
+        val response = client.newCall(GET(searchUrl, headers)).execute()
+
+        return Observable.just(
+            searchMangas(response, query, filters),
+        )
+    }
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request = popularMangaRequest(page)
+
+    override fun searchMangaParse(response: Response): MangasPage = popularMangaParse(response)

```

Comment:
you don't need to override fetchSearchManga here

Reply:
Hello, is there a way to get the query and filters in `searchMangaParse`? I used `fetchSearchManga` because it has the query and filters to pass to the `searchMangas` function.

Reply:
it's fine then to leave as is